intercom-common/src/generators/idl.rs

```diff
@@ -183,6 +198,69 @@ impl IdlModel {
     }
 }
 
+impl<'s> IdlTypeInfo<'s> {
+
+    /// Gets the name of a custom type for IDL.
+    fn get_idl_name_for_custom_type(
+        krate : &ComCrate,
+        ty_name : &str
+    ) -> String {
+
+        let itf = if let Some( itf ) = krate.interfaces().get( ty_name ) {
+            itf
+        } else {
+            return ty_name.to_owned()
+        };
+
+        if itf.item_type() == ::utils::InterfaceType::Struct {
+            format!( "I{}", itf.name() )
+        } else {
+            ty_name.to_owned()
+        }
+    }
+}
+
+impl<'s> IdlTypeInfo<'s> for TypeInfo<'s> {
+
+    fn to_idl(
+        &self
+    ) -> String {
+
+        let type_name = self.get_idl_type_name();
+        let const_specifier = if self.is_mutable || self.pass_by == PassBy::Value { "" } else { "" };
```

Comment:
Either this is the most obscure way to set `const_specifier` to `""`, or one of the branches is missing a `"const"` :)

Reply:
Our current interface didn't use the const in the end. I'll comment that out.

I think we want to omit the const from the default / COM like interface and add appropriate const modifiers for the C++ like interface, including const modifiers for the actual interface functions. We can easily deduce them from the &self parameters.

Reply:
> I think we want to omit the const from the default / COM like interface

Why though? If I remember right, the `const` modifiers in the IDL will end up as `const` in the generated C++ types. Having the `const` in the IDL would further be important once we start respecting Rust's borrow rules.

Essentially calling a non-const method while another non-const method was being called would be in violation of borrowing rules. Alternatively we could force all Rust APIs to be `&self` instead of `&mut self` and force the Rust code to use `RefCell`s, etc. for mutability, thus leaving the conflict resolution up to the user code.

I would prefer some way to communicate this to the caller even in the IDL use case. Even if we can't prevent double non-const method calls in C++, we would give the user the information they need to do recognize that situation manually.

In any case, while we could just ignore all of this and cast the `*mut self_ptr`s into `&mut self`s without having any compiler warnings, this will result in undefined behavior in Rust.